Add Panther Labs integration documentation to README (#9)
panther-labs/grimoire: README.md

> Grimoire is a "REPL for detection engineering" that allows you to generate datasets of cloud audit logs for common attack techniques. It currently supports AWS.

Reviewer: ```suggestion
[Grimoire](https://github.com/dataDog/grimoire) is an interactive shell ("REPL") for detection engineering that allows you to generate datasets of cloud audit logs for common attack techniques. It currently supports AWS.
```

Reviewer: also, titled this section "Panther-enhanced fork of Grimoire" to be consistent with the [SRT changes](https://github.com/panther-labs/stratus-red-team/blob/5d9835a88acbb9239e1be74778feb2ce97520695/README.md)

> ## Panther Labs Integration
> 
> This fork enhances Grimoire with **end-to-end integration testing for detection engineering**. Instead of relying on crafted test cases that may not match real-world log patterns, Grimoire generates authentic attack data and validates your entire detection pipeline.
> 
> > 📺 **Learn More**: Watch the talk "Incorporating End to End Integration Testing into your Detection Engineering Workflow" at [Open Cloud Security Conference](https://youtu.be/4Ijyc2JW-3w) (short version) or BSides Boulder (full version - link TBD)

Reviewer: ```suggestion
```

Reviewer: moved all this content elsewhere

### Key Benefits

> - **Real Attack Simulation**: Uses [Stratus Red Team](https://github.com/panther-labs/stratus-red-team) (Panther fork) to generate genuine log patterns, not synthetic test data

Reviewer: ```suggestion
- **Real attack simulation**: Uses the [Panther Stratus Red Team fork](https://github.com/panther-labs/stratus-red-team) to generate genuine log patterns, not synthetic test data
```

> - **Complete Pipeline Testing**: Validates detection logic AND alerting infrastructure from log ingestion to alert delivery

Reviewer: ```suggestion
- **Complete pipeline testing**: Validates detection logic _and_ alerting infrastructure, from log ingestion to alert delivery
```

> - **Alert Correlation**: Correlates CloudTrail events with Panther security alerts to verify detection coverage

Reviewer: ```suggestion
- **Alert correlation**: Correlates CloudTrail events with Panther security alerts to verify detection coverage
```

> - **Technique Validation**: Confirms whether attack techniques trigger expected Panther detections

Reviewer: ```suggestion
- **Technique validation**: Confirms whether attack techniques trigger expected Panther detections

> 📺 **Learn more**: Watch the "Incorporating End to End Integration Testing into your Detection Engineering Workflow" talk given at [Open Cloud Security Conference](https://youtu.be/4Ijyc2JW-3w) (short version) or BSides Boulder (full version - link TBD)
```

> ### Panther Integration Features

Reviewer: ```suggestion
## Panther integration features
```

> **New Log Source**: Query Panther's data lake via GraphQL API instead of AWS CloudTrail

Reviewer: ```suggestion
### New log source
Query Panther's data lake via GraphQL API instead of AWS CloudTrail:
```

- Retrieve both raw events and security alerts in one operation
- Access extended retention through Panther's data lake
- Leverage existing Panther infrastructure

> **Configuration**:

Reviewer: ```suggestion
### Configuration
```

- `--log-source panther` - Use Panther backend
- `--panther-api-host` - Your Panther GraphQL endpoint  
- `--panther-api-token` - API authentication token

> - `--panther-table` - Table to query (default: "aws_cloudtrail")

Reviewer: I kinda feel like these config flags could go into the "Usage with Panther" section below? they're not screaming "Panther Integration Features" to me

Reviewer: I just suggested renaming "Usage with Panther"" to "Configuration setup"
